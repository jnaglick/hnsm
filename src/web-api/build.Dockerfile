FROM node:14
WORKDIR /web-api
CMD npm ci --no-optional && npm run build
