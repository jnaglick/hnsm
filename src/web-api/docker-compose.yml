version: "3.9"
services:
  web-api-build:
    build:
      context: .
      dockerfile: build.Dockerfile
    image: web-api-build
    volumes:
      - ./:/web-api
