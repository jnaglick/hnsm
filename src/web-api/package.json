{
  "name": "@hnsm/web-api",
  "version": "0.0.0",
  "description": "",
  "scripts": {
    "build": "webpack ./src/index.lambda.ts --mode production",
    "clean": "rm -rf node_modules/ && rm -rf dist/",
    "install:ci": "npm ci --no-optional",
    "start:dev": "webpack ./src/index.local-lambda-emulator.ts --watch --mode development",
    "test": "jest --config=../../jest.config.js --verbose --rootDir=."
  },
  "author": "",
  "license": "NONE",
  "dependencies": {
    "aws-lambda": "1.0.7"
  },
  "devDependencies": {
    "@types/aws-lambda": "8.10.88",
    "@types/jest": "27.0.3",
    "jest": "27.4.5",
    "nodemon-webpack-plugin": "4.7.0",
    "ts-jest": "27.1.2",
    "ts-loader": "9.2.6",
    "tsconfig-paths-webpack-plugin": "3.5.2",
    "typescript": "4.4.4",
    "webpack": "5.59.1",
    "webpack-cli": "4.9.1"
  }
}
