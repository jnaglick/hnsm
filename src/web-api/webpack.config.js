const path = require("path");
const NodemonPlugin = require("nodemon-webpack-plugin");
const TsconfigPathsPlugin = require("tsconfig-paths-webpack-plugin");

const clientConfig = {
  output: {
    path: path.resolve("./dist"),
    filename: "bundle.js",
    libraryTarget: "commonjs",
  },

  resolve: {
    extensions: [".ts", ".js"],
    modules: [
      path.resolve("./node_modules"),
      path.resolve("./src"),
      path.resolve("../common/node_modules"),
    ],
    plugins: [new TsconfigPathsPlugin()],
  },

  module: {
    rules: [
      {
        test: /\.ts$/,
        use: "ts-loader",
        exclude: /node_modules/,
      },
    ],
  },

  plugins: [new NodemonPlugin()],

  target: "node",

  watchOptions: {
    ignored: /node_modules/,
  },
};

module.exports = clientConfig;
