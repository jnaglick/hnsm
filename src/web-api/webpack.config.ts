import * as path from 'path';
import NodemonPlugin from 'nodemon-webpack-plugin';
import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin';

export const WEB_API_DIST_PATH = path.resolve(__dirname, "./dist");

const config = {
  output: {
    path: WEB_API_DIST_PATH,
    filename: "bundle.js",
    libraryTarget: "commonjs",
  },

  resolve: {
    extensions: [".ts", ".js"],
    modules: [
      path.resolve("./node_modules"),
      path.resolve("./src"),
      path.resolve("../common/node_modules"),
    ],
    plugins: [new TsconfigPathsPlugin()],
  },

  module: {
    rules: [
      {
        test: /\.ts$/,
        use: "ts-loader",
        exclude: /node_modules/,
      },
    ],
  },

  plugins: [new NodemonPlugin()],

  target: "node",

  watchOptions: {
    ignored: /node_modules/,
  },
};

export default config;