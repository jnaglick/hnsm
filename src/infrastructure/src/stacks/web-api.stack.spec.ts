import * as cdk from "@aws-cdk/core";
import { SynthUtils } from "@aws-cdk/assert";

import { WebApiStack } from "./web-api.stack";

describe("WebApiStack", () => {
  it("should match snapshot", () => {
    // arrange
    const stack = new WebApiStack(
      new cdk.App({
        context: {
          stage: "test",
        },
      }),
      "WebApiStack"
    );

    // act
    const snapshot = SynthUtils.toCloudFormation(stack);

    // assert
    expect(snapshot).toMatchSnapshot();
  });
});
