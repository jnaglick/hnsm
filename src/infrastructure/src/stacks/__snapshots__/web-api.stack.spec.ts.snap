// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebApiStack should match snapshot 1`] = `
Object {
  "Outputs": Object {
    "ApiGatewayEndpoint5AA8EC3A": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "ApiGateway11E7F47B",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "ApiGatewayDeploymentStageprod1C6D5CD6",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40ArtifactHashC43889A2": Object {
      "Description": "Artifact hash for asset \\"bfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40\\"",
      "Type": "String",
    },
    "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40S3Bucket14B430CD": Object {
      "Description": "S3 bucket for asset \\"bfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40\\"",
      "Type": "String",
    },
    "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40S3VersionKey9EDD4197": Object {
      "Description": "S3 key for asset version \\"bfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "ApiGateway11E7F47B": Object {
      "Properties": Object {
        "Name": "ApiGateway",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ApiGatewayAccountBC7F34C1": Object {
      "DependsOn": Array [
        "ApiGateway11E7F47B",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "ApiGatewayCloudWatchRoleA41EF9FE",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "ApiGatewayCloudWatchRoleA41EF9FE": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiGatewayDeploymentA26796E8d2afc706339d260d7bedf2753da7a818": Object {
      "DependsOn": Array [
        "ApiGatewayGET25EBFEA3",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "ApiGateway11E7F47B",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ApiGatewayDeploymentStageprod1C6D5CD6": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "ApiGatewayDeploymentA26796E8d2afc706339d260d7bedf2753da7a818",
        },
        "RestApiId": Object {
          "Ref": "ApiGateway11E7F47B",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ApiGatewayGET25EBFEA3": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "RequestTemplates": Object {
            "application/json": "{ \\"statusCode\\": \\"200\\" }",
          },
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "LambdaD247545B",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "ApiGateway11E7F47B",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "ApiGateway11E7F47B",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiGatewayGETApiPermissionTestWebApiStackApiGateway76EB5B35GET5A7F9043": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "LambdaD247545B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiGateway11E7F47B",
              },
              "/test-invoke-stage/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiGatewayGETApiPermissionWebApiStackApiGateway76EB5B35GET387793B3": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "LambdaD247545B",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiGateway11E7F47B",
              },
              "/",
              Object {
                "Ref": "ApiGatewayDeploymentStageprod1C6D5CD6",
              },
              "/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LambdaD247545B": Object {
      "DependsOn": Array [
        "LambdaServiceRoleA8ED4D3B",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40S3Bucket14B430CD",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40S3VersionKey9EDD4197",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersbfcdf88b508891eda9947e2934641233da7bfb8bbad45633eb741525ef914a40S3VersionKey9EDD4197",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "bundle.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaServiceRoleA8ED4D3B",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaServiceRoleA8ED4D3B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
